src/util/font.cpp

```diff
@@ -67,15 +67,13 @@ void FontUtils::initializeFonts(const QString& resourcePath) {
         return;
     }
 
-    QList<QFileInfo> files = fontsDir.entryInfoList(
+    const QList<QString> files = fontsDir.entryList(
             QDir::NoDotAndDotDot | QDir::Files | QDir::Readable);
-    for (const QFileInfo& file : files) {
-        const QString& path = file.filePath();
-
+    for (const QString& path : files) {
         // Skip text files (e.g. license files). For all others we let Qt tell
         // us whether the font format is supported since there is no way to
         // check other than adding.
-        if (path.endsWith(".txt", Qt::CaseInsensitive)) {
+        if (path.endsWith(QStringLiteral(".txt"), Qt::CaseInsensitive)) {
```

Comment:
The original char[] verison is faster. Because the compare happens in the char domain.

Reply:
Potentially yes, but the bare char literal is actually even slower because it gets implicitly converted to a QString at runtime. The proper solution is
```suggestion
        if (path.endsWith(QLatin1StringView(".txt"), Qt::CaseInsensitive)) {
```
(according to compiler explorer](https://compiler-explorer.com/z/9sehWz55K)

Digging through the Qt source, the comparision eventually happens in the correct overload of `ucstricmp`:
https://github.com/qt/qtbase/blob/182604ae9ce5fac6c1c79bb3f21b5d50e6da7258/src/corelib/text/qstring.cpp#L1466-L1494

I simply can't judge which is faster in this case without measuring. I doubt it makes much of a difference. Regardless, the plain `".txt"` literal is worse than `QStringLiteral` and `QLatin1StringView`.

Reply:
Ah I see, I was wrong. Qt has only the "faster" `QLatin1StringView`overload for cases where a Latin1 string is already in memory. This saves the runtime conversion to UTF-16. Since `QStringLiteral`we no longer have such a conversion in the literal case. In this case it saves the  
> expand to UTF-16 via unpacking
https://github.com/qt/qtbase/blob/182604ae9ce5fac6c1c79bb3f21b5d50e6da7258/src/corelib/text/qstring.cpp#L588-596
As well. 

The compiler explorer results with a plain char array are surprising. Why is QString::fromUtf8 used. 
For my understanding plain char literals are ASCII and if we want UTF8 we need to add u8"... "

Reply:
It was: fromAscii in QT5 
https://github.com/qt/qtbase/blob/83f5c3c26a59849197bd8d3a661afc87893866ba/src/corelib/text/qstring.h#L807
now: fromUtf8
https://github.com/qt/qtbase/blob/182604ae9ce5fac6c1c79bb3f21b5d50e6da7258/src/corelib/text/qstring.h#L835

Reply:
So you agree that the `QStringLiteral` is correct?

Reply:
Yes